sdk/blockchain-client-provider/src/implementation/BlockchainClientProvider.ts

```diff
@@ -38,38 +40,31 @@ export class BlockchainClientProvider implements IBlockchainClientProvider {
   /** PUBLIC */
 
   /** @see IBlockchainClientProvider.getBlockchainClient */
-  public getBlockchainClient(params: { chainInfo: IChainInfo }): IBlockchainClient {
-    const provider = this._blockchainClients[params.chainInfo.chainId]
-    if (!provider) {
-      throw new Error('Provider not found')
-    }
-    return provider
-  }
-
-  /** @see IBlockchainClientProvider.getCustomBlockchainClient */
-  public getCustomBlockchainClient(params: {
-    rpcUrl: string
+  public getBlockchainClient(params: {
     chainInfo: IChainInfo
+    rpcUrl?: string
   }): IBlockchainClient {
-    const chain = defineChain({
-      id: params.chainInfo.chainId,
-      name: params.chainInfo.name,
-      nativeCurrency: {
-        decimals: 18,
-        name: 'Ether',
-        symbol: 'ETH',
-      },
-      rpcUrls: {
-        default: {
-          http: [params.rpcUrl],
+    const chain = this._chains[params.chainInfo.chainId]
+    assert(chain, 'Blockchain client was found but not its chain, this should never happen')
```

Comment:
I think the message is not precise, should be something like "chainId not supported" as _chains should contain all supported chains

Reply:
TBH  it's just my personal style so feel free to skip that really  :)

Reply:
I've actually seen that I got it wrong: my intention was to make sure that both elements from `blockchainClients` and `_chains` are available (it would be a programming bug if they are not) but the assert is in the wrong place. I've moved it below and added an exception